ui/src/app/(index)/page.tsx

```diff
@@ -1,7 +1,79 @@
-const Home = () => {
+import Image from 'next/image';
+import BeforeLogin from '@/components/home/BeforeLogin';
+import AfterLogin from '@/components/home/AfterLogin';
+import {getCurrentUser} from '@/access/AuthenticationService';
+import Role from '@/access/Role';
+import LoginButton from '@/components/home/LoginButton';
+import {getAnnouncements, getNumberOfGroupings, getNumberOfMemberships} from '@/services/GroupingsApiService';
+import React from 'react';
+import Announcement from '@/components/home/Announcement';
+
+const Home = async () => {
+    const [currentUser, numberOfGroupings, numberOfMemberships, announcements] = await Promise.all([
+        getCurrentUser(),
+        getNumberOfGroupings(),
+        getNumberOfMemberships(),
+        getAnnouncements()
+    ]);
+
+    const handleAnnouncements = () => {
+        if (Object.prototype.hasOwnProperty.call(announcements, 'announcements')) {
+            return announcements.announcements
+                .filter((announcement) => announcement.state !== 'Active')
+                .map((announcement) => announcement.message);
+        } else {
+            return [];
+        }
+    };
```

Comment:
I realize now that setting this to a variable instead of function would be better. A good variable name would be `activeAnnouncements`. Also you should not need the Object.prototype.hasOwnProperty, The announcements should be of Announcement type so that you can directly just do `announcements.announcements`.

Reply:
done